barcode/templates/barcode_client.html

```diff
@@ -0,0 +1,246 @@
+  var isScannerDetecting = false;
+
+  var scannerDetectionBadgeNumHandlers = [];
+  var scannerDetectionUUIDHandlers = [];
+  var scannerDetectionOnCompleteHandlers = [];
+  var scannerDetectionOnErrorHandlers = [];
+  var scannerDetectionOnReceiveHandlers = [];
+
+  $(function () {
+    $(document).scannerDetection({
+      timeBeforeScanTest: 100,
+      avgTimeByChar: 30,
+      endChar: [9, 13, '\\'],
+      startChar: [BADGE_NUM_CHAR, '\\'],
+      preventDefault: false,
+      stopPropagation: false,
+      onComplete: function(barcode, qty) {
+        barcode = ('' + barcode).trim();
+
+        // Remove non-word characters bracketing the barcode.
+        // Some scanners add these but we don't want them.
+        var first = barcode.substr(0, 1),
+            last = barcode.substr(barcode.length - 1);
+        if (first === last && last.match(/\W/g)) {
+          barcode = barcode.substr(1, barcode.length - 2);
+        }
+
+        if (scannerDetectionUUIDHandlers.length > 0) {
+          var uuidMatch = barcode.match(RE_UUID);
+          if (uuidMatch) {
+            $.each(scannerDetectionUUIDHandlers, function(i, handler) {
+              handler({barcode: barcode, uuid: uuidMatch[1]});
+            });
+          }
+        }
+
+        if (scannerDetectionBadgeNumHandlers.length > 0) {
+          getBadgeNumFromBarcode(barcode, function (response) {
+            $.each(scannerDetectionBadgeNumHandlers, function(i, handler) {
+              handler(response);
+            });
+          });
+        }
+
+        $.each(scannerDetectionOnCompleteHandlers, function(i, handler) {
+          handler({barcode: barcode});
+        });
+      },
+      onError: function (key) {
+        $.each(scannerDetectionOnErrorHandlers, function(i, handler) {
+          handler(key);
+        });
+      },
+      onReceive: function(event) {
+        $.each(scannerDetectionOnReceiveHandlers, function(i, handler) {
+          handler(event);
+        });
+      }
+    });
+  });
+
+  (function($) {
+    $.fn.barcodeField = function(options) {
+      // Default settings, overridden by options
+      var settings = $.extend({
+        // The target field for auto-fill and focus/blur events, if different
+        // from the element used to initialize the barcodeField.
+        targetField: null,
+
+        // Control how the target field is blurred/focused.
+        blurOnKeys: [],
+        refocusAfterBlurOnKeys: true,  // Refocuses if blurred from blurOnKeys
+        focusOnError: true,  // Always focus, regardless of blurOnKeys
+        focusOnComplete: true,  // Always focus, regardless of blurOnKeys
+
+        // Turn off toastr error messages entirely.
+        showErrorMessage: true,
+
+        // Turn off toastr error messages, but only when target field is hidden.
+        showErrorMessageWhenHidden: true,
+
+        // Determine if we do any special data handling.
+        // detectBadgeNum and detectUUID are mututally exclusive. If both are
```

Comment:
Minor nitpick: it's easier to read comments if they don't split across lines like this. I THINK `// If both true, then only detectBadgeNum is used.` will fit on one line?